src/pages/trips/[id].tsx

```diff
@@ -2,17 +2,58 @@ import React from 'react'
 import SplitLayout from '../../components/splitLayout'
 import Itinerary from '../../components/itinerary'
 import Map from '../../components/map'
+import { prisma } from '../../server/db/client'
+import { type GetServerSideProps } from 'next'
 
+interface IItineraryData {
+  coverPhoto?: string
+  destinations: string[]
+  endDate: Date
+  id: number
+  likes: number
+  name: string
+  public: boolean
+  profileId: number
+  startDate: Date
+  tripDays: ITripDay
+}
+
+interface ITripDay {
+  id: number,
+  date: Date
+  itineraryId: number
+}
 
-const tripPage = () => {
+const tripPage = (itineraryData: IItineraryData) => {
+
+    console.log('itineraryData:',  itineraryData)
 
   return (
-    <SplitLayout leftChildren={<Itinerary />} rightChildren={<Map />}/>
+    <SplitLayout leftChildren={<Itinerary itin={itineraryData} />} rightChildren={<Map />}/>
   )
 }
 
 export default tripPage
 
-// export async function getServerSideProps() {
-//    await axios.get('/api/itinerary/get')
-// }
\ No newline at end of file
+export const getServerSideProps: GetServerSideProps = async ({query, req, res}) => {
+
+  let itineraryData;
+
+  try {           
+    const data = await prisma.itinerary.findUnique({
+      where: {
+        id: Number(query.id),
+      },
+      include: {
+        tripDays: true
```

Comment:
Need to nest one layer further and also get the corresponding activities. Ex. tripdays: { activities: true }